code/modules/mob/mob_helpers.dm

```diff
@@ -224,28 +222,36 @@
   *
   * It's fairly easy to fix if dealing with single letters but not so much with compounds of letters./N
   */
-/proc/ninjaspeak(n) //NINJACODE
-	var/te = html_decode(n)
-	var/t = ""
-	n = length(n)
-	var/p = 1
-	while(p <= n)
-		var/n_letter
-		var/n_mod = rand(1,4)
-		if(p+n_mod>n+1)
-			n_letter = copytext(te, p, n+1)
-		else
-			n_letter = copytext(te, p, p+n_mod)
-		if (prob(50))
-			if (prob(30))
-				n_letter = text("[n_letter]-[n_letter]-[n_letter]")
+/proc/ninjaspeak(phrase) //NINJACODE
+	phrase = html_decode(phrase)
+	var/leng = length(phrase)
+	. = ""
+	var/stutterphrase = ""
+	var/rawchar = ""
+	var/n_mod = rand(1, 4)
+	for(var/i = 1, i <= leng, i += length(rawchar))
+		rawchar = phrase[i]
+		if(rawchar == " " || i == leng)
+			n_mod = -1
+		if(n_mod > 0)
+			n_mod--
+			stutterphrase += rawchar
+			continue
+		if(rawchar != " ")
+			stutterphrase += rawchar
+
+		n_mod = rand(1, 4)
+		if(stutterphrase && prob(50))
+			if(prob(30))
+				stutterphrase = "[stutterphrase]-[stutterphrase]-[stutterphrase]"
 			else
-				n_letter = text("[n_letter]-[n_letter]")
-		else
-			n_letter = text("[n_letter]")
-		t = text("[t][n_letter]")
-		p=p+n_mod
-	return copytext(sanitize(t),1,MAX_MESSAGE_LEN)
+				stutterphrase = "[stutterphrase]-[stutterphrase]"
+		if(rawchar == " ")
+			stutterphrase += rawchar
+
+		. += stutterphrase
+		stutterphrase = ""
+	return sanitize(.)

```

Comment:
this is actually unused, just kill it.

Reply:
Are you serious? Goddamnit I spent the most time on this one.